addons/waterways/plugin.gd

```diff
@@ -221,27 +221,28 @@ func _forward_3d_gui_input_river(camera: Camera3D, event: InputEvent) -> int:
 			if closest_segment == -1:
 				var end_pos = _edited_node.curve.get_point_position(_edited_node.curve.get_point_count() - 1)
 				var end_pos_global : Vector3 = _edited_node.to_global(end_pos)
-					
+
 				var z : Vector3 = _edited_node.curve.get_point_out(_edited_node.curve.get_point_count() - 1).normalized()
 				var x := z.cross(Vector3.DOWN).normalized()
 				var y := z.cross(x).normalized()
 				var _handle_base_transform = Transform3D(
 					Basis(x, y, z) * global_transform.basis,
 					end_pos_global
 				)
-			
+
 				var plane := Plane(end_pos_global, end_pos_global + camera.transform.basis.x, end_pos_global + camera.transform.basis.y)
 				var new_pos
 				if constraint == RiverControls.CONSTRAINTS.COLLIDERS:
-					var space_state = _edited_node.get_world_3d().direct_space_state
-					var result = space_state.intersect_ray(ray_from, ray_from + ray_dir * 4096)
+					var space_state := _edited_node.get_world_3d().direct_space_state
+					var ray_params := PhysicsRayQueryParameters3D.create(ray_from, ray_from + ray_dir * 4096)
+					var result = space_state.intersect_ray(ray_params)
```

Comment:
`result` needs static typing, which should be a dictionary.

I tell my team to never use `:=` unless explicitly being assigned a type in that line, eg `var dir := Vector3(1,2,3)`. Otherwise, always type it. Especially from all engine returns.